src/include/miopen/tensor_layout.hpp

```diff
@@ -35,6 +35,26 @@
 
 namespace miopen {
 
+enum tensor_layout_t
+{
+    tensor_layout_nchw,
+    tensor_layout_ncdhw,
+    tensor_layout_nhwc,
+    tensor_layout_ndhwc,
+};
```

Comment:
Why don't we use the existing enum for layouts?

Reply:
Let's create a ticket for this so we don't forget

Reply:
Also let's not add things to the library that are only used in the tests

Reply:
Good point @averinevg . I moved it here because I couldn't find a suitable existing file under `test`. I'm moving `tensor_layout_to_string` into a new file, gtest/tensor_layout_util.hpp. PR will be up this evening. `tensor_layout_t` no longer exists, ofc. Any alternate suggestions before I get too far into that work?

Reply:
My point was that you are trying to invent something that already exists. In addition to being a waste of time, it could lead to confusion. Since this will be fixed, I have no complaints. But it’s still better to prevent the presence of such “unfinished” pull requests in the `develop`, especially when the problem is known and the solution is trivial.

Reply:
Oh @averinevg, you missed the history of this. `tensor_layout_t` is not my code, I just refactored it to remove existing duplicates in the tests. The duplication of `miopenTensorLayout_t` was not identified before it was too late in the sprint to justify holding up this PR. So yes, the proper approach is to handle this immediately in a followup PR.